neutromeratio/tautomers.py

```diff
@@ -5,138 +5,304 @@
+        assert(type(final_state_mol) == Chem.Mol)
 
-def from_mol_to_ani_input(mol:Chem.Mol, nr_of_conf:int, in_solvent:bool=False)->dict:
-    """
-    Generates atom_list and coord_list entries from rdkit mol.
-    Parameters
-    ----------
-    mol : rdkit.Chem.Mol
-    nr_of_conf : int
+        self.intial_state_mol:Chem.Mol = intial_state_mol
+        self.final_state_mol:Chem.Mol = final_state_mol
 
-    Returns
-    -------
-    { 'ligand_atoms' : atoms, 'ligand_coords' : coord_list} 
-    """
-    
-    # generate atom list
-    atom_list = []
-    for a in mol.GetAtoms():
-        atom_list.append(a.GetSymbol())
+        intial_state_ani_input = self._from_mol_to_ani_input(self.intial_state_mol)
+        self.intial_state_ligand_atoms = intial_state_ani_input['ligand_atoms']
+        self.intial_state_ligand_coords = intial_state_ani_input['ligand_coords']
+        self.intial_state_ligand_topology:md.Topology = intial_state_ani_input['ligand_topology']
+        self.intial_state_ase_mol:Atoms = intial_state_ani_input['ase_mol']
+
+        final_state_ani_input = self._from_mol_to_ani_input(self.final_state_mol)
+        self.final_state_ligand_atoms = final_state_ani_input['ligand_atoms']
+        self.final_state_ligand_coords = final_state_ani_input['ligand_coords']
+        self.final_state_ligand_topology:md.Topology = final_state_ani_input['ligand_topology']
+        self.final_state_ase_mol:Atoms = final_state_ani_input['ase_mol']
+
+        # attributes for the equilibrium protocol
+        # these values are set by perform_tautomer_transformation_forward
+        # or perform_tautomer_transformation_reverse and correspond either
+        # to the idx of initial in case of forward or final in case of reverse
+        self.hybrid_hydrogen_idx_at_donor_heavy_atom:int = -1
+        self.hybrid_hydrogen_idx_at_acceptor_heavy_atom:int = -1
+        self.hybrid_coords:list = []
+        self.hybrid_ase_mol:Atoms = None
+        self.hybrid_topology:md.Topology = None
+        
+        # attributes for the NCMC protocol
+        self.heavy_atom_hydrogen_donor_idx:int = -1
+        self.hydrogen_idx:int = -1
+        self.heavy_atom_hydrogen_acceptor_idx:int = -1
+        self.ncmc_restraints:list = []
+
+
+    def perform_tautomer_transformation_forward(self):
+        """
+        Performs a tautomer transformation from the initial state to the final state 
+        and sets parameter and restraints using the indexing of the initial state mol.
+        """
+
+        m1 = copy.deepcopy(self.intial_state_mol)
+        m2 = copy.deepcopy(self.final_state_mol)
+        self._perform_tautomer_transformation(m1, m2)
+        self._generate_hybrid_structure(self.intial_state_ligand_atoms, None, self.intial_state_ligand_coords[0], self.intial_state_ligand_topology)
+
+    def perform_tautomer_transformation_reverse(self):
+        """
+        Performs a tautomer transformation from the initial state to the final state 
```

Comment:
Implementation appears right, but docstring is backwards (should be `final` --> `initial`)